### ISSUE TITLE
Linear Maps not working with power of 2 cyclotomics

### ISSUE THREAD

**Issue by benjaminhmtan** [homenc/HElib#247]
Hi, I tried using the linear map generation and evaluation functionality in the EncryptedArray class but found that it does not output the correct result when m is a power of 2, say when m=16384 and p=257. I have not encountered similar problems when I used general cyclotomics.

**Comment by benjaminhmtan** [comment]
Thanks, I have tried that already but L is not the problem.

**Comment by benjaminhmtan** [comment]
So far, I have isolated the problem to the Frobenius automorphism calculations. Here is a small working example that shows the Frobenius automorphism computations are wrong when m is a power of 2.

```
long m = 512;
long p = 31;

context(m, p, 1);
buildModChain(context, 45);

NTL::ZZX F = context.alMod.getFactorsOverZZ()[0];
EncryptedArray ea(context, F);
long fieldDegree = ea.getDegree();
long numSlots = ea.size();

FHESecKey secretKey(context);
const FHEPubKey &publicKey = secretKey;
secretKey.GenSecKey(64);
addFrbMatrices(secretKey);

NTL::ZZ_p::init(ZZ(p));
NTL::ZZ_pX G;
G = NTL::conv<NTL::ZZ_pX>(F);
NTL::ZZ_pE::init(G);

NTL::ZZ_pX monomial;
NTL::SetCoeff(monomial, 1);
NTL::ZZ_pE monomialPE = NTL::conv<NTL::ZZ_pE>(monomial);

std::vector<NTL::ZZX> tempPlaintext(numSlots, NTL::ZZX(NTL::INIT_MONO, 1));
Ctxt tempCiphertext(publicKey);
ea.encrypt(tempCiphertext, publicKey, tempPlaintext);

std::vector<Ctxt> linMonomials(fieldDegree, tempCiphertext);
for(unsigned long i = 0; i < 3; i++)
{
    linMonomials[i].frobeniusAutomorph(i);
    ea.decrypt(linMonomials[i], secretKey, tempPlaintext);
    std::clog << tempPlaintext[0] << std::endl << power(monomialPE, pow(plaintextModulus, i)) << std::endl;
}

return 0;
```

**Comment by benjaminhmtan** [comment]
Looks to be fixed with the latest helib 1.0.0 (beta) and ntl 11.3.2.
